### ISSUE TITLE
Give users option of disabling `<input type="date">` for Date questions

### ISSUE THREAD

**Issue by neel-a-mehta** [civiform/civiform#1521]
We try to use the native `<input type="date">` HTML element for Date questions, which is usually a good thing, except that [accessibility experts say that this question type isn't fully accessible](https://www.hassellinclusion.com/blog/input-type-date-ready-for-use/), especially for keyboard and voice-input users.

In the applicant's profile page (#1496), we should offer a toggle to let them replace the native `<input type="date">` input elements with plain old `<input type="text">` elements. These `text` inputs are less fancy and therefore a bit less straightforward to use, and the applicant has to be sure that they're getting the syntax just right. But it might be really helpful for users with accessibility needs. 

Put this setting under a potential "Accessibility Settings" heading on the profile page.

If the user has elected to use `text` inputs, we should require them to input dates in the format `MM/DD/YYYY`, just as the native datepicker uses.

## Some problems raised about `<input type="date">`
These are all problems that Google's a11y testers picked up on.

* The contrast in the native calendar picker widget (on Android, no less) isn't high enough. 

![image](https://user-images.githubusercontent.com/77128518/123717947-f6d17100-d84b-11eb-993b-93440459c519.png)

* The UI is inconsistent across browsers
* Focus does not automatically move to the calendar widget once it's opened. Tabbing through the form just skips the calendar widget.
* Screen readers do not announce when the calendar widget has popped up.
* Screen readers do not announce dates and times properly:

> ACTUAL RESULT:
Jaws: Screen reader is announcing month/date/Year as 'mm date time edit/dd date time edit/yyyy date time edit' which can be confusing for a screen reader user.
Voice Over: User is not getting informed that date picker has calendar popup associated with it. Screen reader is announcing the field value as a plain text.
> EXPECTED RESULT: Screen reader should read the calendar info properly. It should be announced month/date/year instead of mm/dd/yyyy.

**Comment by neel-a-mehta** [comment]
## More a11y problems with `<input type="date">`

* Screen readers may not inform applicants when the date value is changed using the date-picker widget.

> ACTUAL RESULT: User is not getting informed upon changing the date value on the stepper.
> EXPECTED RESULT: Screen reader should notify user upon changing the date value on the stepper.

**Comment by rahaghas** [comment]
Instead of using <input type="date">, we could control for what will pop up regardless of browser/platform and use this component: https://material.io/components/date-pickers#desktop-pickers

We should be able to control for the color and make sure it meets accessibility standards. If we choose to use this, I THINK this would be the resource: https://github.com/material-components/material-components-web
